src/managers/OnOfframpManager.sol

```diff
@@ -0,0 +1,132 @@
+
+    bool public permissionlessOnramp;
+    bool public permissionlessOfframp;
+    mapping(address => bool) public manager;
+    mapping(address asset => bool) public onramp;
+    mapping(address asset => mapping(address receiver => bool)) public offramp;
+
+    constructor(PoolId poolId_, ShareClassId scId_, address spoke_, IBalanceSheet balanceSheet_) {
+        poolId = poolId_;
+        scId = scId_;
+        spoke = spoke_;
+        balanceSheet = balanceSheet_;
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // Owner actions
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc IUpdateContract
+    function update(PoolId poolId_, ShareClassId, /* scId */ bytes calldata payload) external {
+        require(poolId == poolId_, InvalidPoolId());
+        require(msg.sender == spoke, NotAuthorized());
+
+        uint8 kind = uint8(UpdateContractMessageLib.updateContractType(payload));
+
+        if (kind == uint8(UpdateContractType.UpdateAddress)) {
+            UpdateContractMessageLib.UpdateContractUpdateAddress memory m =
+                UpdateContractMessageLib.deserializeUpdateContractUpdateAddress(payload);
+            address who = m.who.toAddress();
+
+            if (m.kind == "onramp") {
+                onramp[who] = m.isEnabled;
+                emit UpdateOnramp(who, m.isEnabled);
+            } else if (m.kind == "offramp") {
+                address asset = m.what.toAddress();
+                offramp[asset][who] = m.isEnabled;
+                emit UpdateOfframp(asset, who, m.isEnabled);
+            }
+        } else if (kind == uint8(UpdateContractType.Toggle)) {
+            UpdateContractMessageLib.UpdateContractToggle memory m =
+                UpdateContractMessageLib.deserializeUpdateContractToggle(payload);
+
+            if (m.what == "permissionlessOnramp") permissionlessOnramp = m.isEnabled;
+            else if (m.what == "permissionlessOfframp") permissionlessOfframp = m.isEnabled;
+            emit UpdatePermissionless(m.what, m.isEnabled);
+        } else {
+            revert UnknownUpdateContractType();
+        }
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // Deposit & withdraw actions
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc IDepositManager
+    function deposit(address asset, uint256, /* tokenId */ uint128 amount, address owner) external {
+        require(manager[msg.sender] || permissionlessOnramp, IAuth.NotAuthorized());
+
+        require(onramp[asset], NotAllowedOnrampAsset());
+        require(owner == address(this) || owner == msg.sender);
+        require(amount <= IERC20(asset).balanceOf(owner), InvalidAmount());
+
+        balanceSheet.deposit(poolId, scId, asset, 0, amount);
```

Comment:
Only supports ERC20 assets. That is intentional, I think it's good to keep this simple, and focus on the main use cases (like JTRSY). We can adapt or build a new version later, for ERC6909 assets.